lib/level10/games/games.ex

```diff
@@ -4,110 +4,120 @@ defmodule Level10.Games do
   functions will take in a game struct and manipulate that struct and return
   it.
   """
+  alias Level10.Games.{Game, GameRegistry, Player}
 
-  alias Level10.Games.{Card, Game}
+  @typep game_name :: Agent.name()
 
-  @doc """
-  This function will make all of the necessary changes to start a round. It
-  will increment the current_round, shuffle a new deck of cards as the draw
-  pile, and deal out a new hand to each player.
-  """
-  @spec start_round(Game.t()) :: Game.t()
-  def start_round(game) do
-    game
-    |> increment_current_round()
-    |> put_new_deck()
-    |> put_new_hands()
-  end
-
-  @spec increment_current_round(Game.t()) :: Game.t()
-  defp increment_current_round(%{current_round: current_round} = game) do
-    %{game | current_round: current_round + 1}
-  end
+  @max_attempts 10
 
-  @spec put_new_deck(Game.t()) :: Game.t()
-  defp put_new_deck(game) do
-    %{game | draw_pile: new_deck()}
+  @spec create_game(String.t()) :: {:ok, Game.join_code(), Player.id()} | :error
+  def create_game(player_name) do
+    player = Player.new(player_name)
+    do_create_game(player, @max_attempts)
   end
 
-  @spec put_new_hands(Game.t()) :: Game.t()
-  defp put_new_hands(%{draw_pile: deck, players: players} = game) do
-    {hands, draw_pile} =
-      Enum.reduce(players, {%{}, deck}, fn %{name: player_name}, {hands, draw_pile} ->
-        {player_hand, rest_of_pile} = Enum.split(draw_pile, 10)
-        hands = Map.put(hands, player_name, player_hand)
-        {hands, rest_of_pile}
-      end)
+  @spec do_create_game(Player.t(), non_neg_integer()) ::
+          {:ok, Game.join_code(), Player.id()} | :error
+  defp do_create_game(player, attempts_remaining)
 
-    %{game | draw_pile: draw_pile, hands: hands}
+  defp do_create_game(_player, 0) do
+    :error
   end
 
-  @spec new_deck() :: Card.t()
-  defp new_deck do
-    color_cards =
-      for value <- ~W[one two three four five six seven eight nine ten eleven twelve wild]a,
-          color <- ~W[blue green red yellow]a,
-          card = Card.new(value, color),
-          _repeat <- 1..2 do
-        card
-      end
+  defp do_create_game(player, attempts_remaining) do
+    join_code = Game.generate_join_code()
 
-    skips = for _repeat <- 1..4, do: Card.new(:skip, :blue)
+    case Agent.start_link(Game, :new, [join_code, player], name: via(join_code)) do
```

Comment:
Hmm I don't think we want to link the agent to the current process right? If we're calling this from the live view, that would mean if the host loses their connection the game would die too right?

Reply:
Right
There's no need to supervise them, right? We should just do an `Agent.start/4`?